benchmarks/envoy_filter_fuzz_test/build.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash -eu
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+export CFLAGS="$CFLAGS"
+export CXXFLAGS="$CXXFLAGS"
+
+declare -r FUZZER_TARGETS_CC=$(find . -name *_fuzz_test.cc)
+declare -r FUZZER_TARGETS="$(for t in ${FUZZER_TARGETS_CC}; do echo "${t:2:-3}"; done)"
+
+FUZZER_DICTIONARIES="\
+"
+
+# Copy $CFLAGS and $CXXFLAGS into Bazel command-line flags, for both
+# compilation and linking.
+#
+# Some flags, such as `-stdlib=libc++`, generate warnings if used on a C source
+# file. Since the build runs with `-Werror` this will cause it to break, so we
+# use `--conlyopt` and `--cxxopt` instead of `--copt`.
+#
+declare -r EXTRA_BAZEL_FLAGS="$(
+for f in ${CFLAGS}; do
+  echo "--conlyopt=${f}" "--linkopt=${f}"
+done
+for f in ${CXXFLAGS}; do
+  echo "--cxxopt=${f}" "--linkopt=${f}"
+done
+
+if [ "$SANITIZER" = "undefined" ]
```

Comment:
We don't set SANITIZER, i think you should check using undefined in CFLAGS or set this always.

Reply:
We probably should then right?

Reply:
I did modify it in order to recognize if undefined is used in -fsanitize, but I guess setting it would be a good idea. Still, what if 2 sanitizers are used, such as ASAN and UBSAN? $SANITIZER would be "address, undefined"

Reply:
If we set it at all then maybe it would be `address, undefined`. We don't combine sanitizers in OSS-Fuzz.

Another issue we need to figure out is how we will use UBSAN.
In OSS-Fuzz, we don't actually do -fsanitize=undefined. We specify specific ubsan checks so the flags look more like this:
`-fsanitize=unsigned-overflow,signed-overflow,bool`

Reply:
I see...so we use not sanitizers at all in fuzzbench, or are there some default ones? If we use UBSAN with the flags instead of undefined, that might also be a reason why envoy is not building, since I am looking for the "undefined" sanitizer. If we don't do that in FuzzBench, though, I guess it means our only clue right now is the libprotobuf-mutator thing

Reply:
>I see...so we use not sanitizers at all in fuzzbench, or are there some default ones

I don't exactly understand this.
We don't use all of the default UBSAN checks in OSS-Fuzz.

>if we use UBSAN with the flags instead of undefined

In Fuzzbench, I don't think we are using *any* ubsan flags yet.

>If we don't do that in FuzzBench, though, I guess it means our only clue right now is the libprotobuf-mutator thing

Actually I don't really consider this a clue. AFL builds fine with libprotobuf-mutator targets. Running them together also doesn't result in an error. But it's useless to run AFL with them. It's a runtime issue and shouldn't affect builds.

I think the best clue we have is simply that envoy is a very finnicky project. When I think of projects with build problems in OSS-Fuzz envoy is probably the first that comes to mind. It's constantly having them.